ims/einstein/iscsi/tgt.py

```diff
@@ -11,180 +12,154 @@
 
 
 class TGT(ISCSI):
-    '''
-        Class for implementing TGT
-    '''
+    """ Class for implementing TGT """
 
-    # We should get all the config related to ceph in here.
-    # Also, root_password is something that can be avoided.
-
-    # Also removed sudo everywhere since it was causing issues
-    def __init__(self, fs_config_loc, fs_user, root_password):
-        self.arglist = ["service", "tgtd"]
+    def __init__(self, fs_config_loc, fs_user, fs_pool):
         self.TGT_ISCSI_CONFIG = "/etc/tgt/conf.d/"
         self.fs_config_loc = fs_config_loc
         self.fs_user = fs_user
-        self.root_password = root_password
+        self.fs_pool = fs_pool
 
     @log
     def start_server(self):
-        '''
-        Have to parse the output and send a status code.
-        :return:
-        '''
-        # arglist = self.arglist
-        # arglist.append("start")
-        # Need to fix shell=True everywhere
-        command = "service tgtd start"
-        proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-        output, err = proc.communicate(self.root_password +
-                                       "\n")
-        logger.debug("Output = %s, Error = %s", output, err)
-        if self.show_status() is 'Running':
+        """
+        Starts the tgt service
+
+        :return: None
+        """
+        try:
+            command = "service tgtd start"
```

Comment:
I am not against moving this part but what I am telling is starting and stopping a server are logically part of that iscsi server, so if we move them to shell.py, what if there is an iscsi server that starts and stops using some Python script, should we have another Python.py for that or wrap it in shell.py? I think I am OK with moving as long as start and stop server are available as methods in TGT or iscsi but call the ones in shell.py or implement what @knikolla suggested for now.